Building docker image on Windows 11 failed at "/bin/sh -c make install -j4"

Platform Windows 11Pro
Docker version 20.10.17, build 100c701
Execution command: cd docker ; docker build .
Error log:
```
#37 5.598 /usr/local/include/highfive/bits/H5Converter_misc.hpp:83:19: error: static assertion failed: Booleans are not supported yet.
#37 5.598    83 |     static_assert(!std::is_same<type, bool>::value, "Booleans are not supported yet.");
#37 5.598       |                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#37 8.713 /usr/src/bioslam/src/mathutils.cpp: In function 'Eigen::VectorXd mathutils::angleUnwrapOnKnownDomain(const VectorXd&, const double&, const double&, const double&)':
#37 8.713 /usr/src/bioslam/src/mathutils.cpp:410:49: warning: unused variable 'domainCenter' [-Wunused-variable]
#37 8.713   410 |         double domainWidth=domainMax-domainMin, domainCenter=(domainMax+domainMin)/2;
#37 8.713       |                                                 ^~~~~~~~~~~~
#37 8.792 /usr/src/bioslam/src/mathutils.cpp: In function 'Eigen::VectorXd mathutils::adjustRegressionSlopeToTargetSlope(const VectorXd&, const VectorXd&, const double&)':
#37 8.792 /usr/src/bioslam/src/mathutils.cpp:452:40: warning: unused variable 'tol' [-Wunused-variable]
#37 8.792   452 |         double newSlope, newIntercept, tol=1.0e-10;
#37 8.792       |                                        ^~~
#37 9.074 In file included from /usr/local/include/highfive/bits/H5Attribute_misc.hpp:22,
#37 9.074                  from /usr/local/include/highfive/bits/H5Annotate_traits_misc.hpp:18,
#37 9.074                  from /usr/local/include/highfive/H5File.hpp:94,
#37 9.074                  from /usr/local/include/highfive/H5Easy.hpp:58,
#37 9.074                  from /usr/src/bioslam/src/lowerBodyPoseEstimator.cpp:25:
#37 9.074 /usr/local/include/highfive/bits/H5ReadWrite_misc.hpp: At global scope:
#37 9.074 /usr/local/include/highfive/bits/H5ReadWrite_misc.hpp:106:1: error: 'HighFive::details::BufferInfo<T>::BufferInfo(const HighFive::DataType&, F) [with F = HighFive::SliceTraits<Derivate>::write(const T&) [with T = bool; Derivate = HighFive::DataSet]::<lambda()>; T = bool]', declared using local type 'HighFive::SliceTraits<Derivate>::write(const T&) [with T = bool; Derivate = HighFive::DataSet]::<lambda()>', is used but never defined [-fpermissive]
#37 9.074   106 | BufferInfo<T>::BufferInfo(const DataType& dtype, F getName)
#37 9.074       | ^~~~~~~~~~~~~
#37 9.490 make[2]: *** [CMakeFiles/bioslam.dir/build.make:63: CMakeFiles/bioslam.dir/src/lowerBodyPoseEstimator.cpp.o] Error 1
#37 9.490 make[2]: *** Waiting for unfinished jobs....
#37 10.32 /usr/src/bioslam/src/gtsamutils.cpp: In function 'uint gtsamutils::nearestIdxToVal(std::vector<double>, double)':
#37 10.32 /usr/src/bioslam/src/gtsamutils.cpp:189:16: warning: 'nearestIdx' may be used uninitialized in this function [-Wmaybe-uninitialized]
#37 10.32   189 |         return nearestIdx;
#37 10.32       |                ^~~~~~~~~~
#37 15.54 make[1]: *** [CMakeFiles/Makefile2:173: CMakeFiles/bioslam.dir/all] Error 2
#37 15.54 make: *** [Makefile:141: all] Error 2
#37 ERROR: executor failed running [/bin/sh -c make install -j4]: exit code: 2
------
 > [33/33] RUN make install -j4:
------
executor failed running [/bin/sh -c make install -j4]: exit code: 2

```

--- a/.github/workflows/dockerfile-build.yml
+++ b/.github/workflows/dockerfile-build.yml
@@ -1,0 +1,20 @@
+name: Dockerfile
+
+on: [push, pull_request]
+
+jobs:
+  test:
+    name: Build, test Dockerfile
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v3
+
+    - name: Build and run docker image
+      run: |
+        docker build -t bioslam -f ./docker/Dockerfile .
+        docker run -id --name bioslam bioslam
+
+    - name: Run tests
+      run: docker exec -i bioslam make test

--- a/README.md
+++ b/README.md
@@ -1,6 +1,7 @@
-# Bioslam: IMU-Based Human Skeletal Pose Estimation ![Built, Test CI](https://github.com/tmcg0/bioslam/actions/workflows/build-test-linux.yml/badge.svg)
-
-**Author:** [Tim McGrath](https://www.researchgate.net/profile/Tim_Mcgrath9)
+<h1 align="center">
+  Bioslam: IMU-Based Human Skeletal Pose Estimation<br>
+  <a href="https://github.com/tmcg0/bioslam/actions/workflows/build-test-linux.yml/badge.svg"><img src="https://github.com/tmcg0/bioslam/actions/workflows/build-test-linux.yml/badge.svg"/></a> <a href="https://github.com/tmcg0/bioslam/actions/workflows/dockerfile-build.yml/badge.svg"><img src="https://github.com/tmcg0/bioslam/actions/workflows/dockerfile-build.yml/badge.svg"/></a>
+</h1>
 
 ## What is Bioslam?
 

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -28,7 +28,7 @@ WORKDIR /usr/src/
 RUN git clone --single-branch --branch 3.3.9 https://gitlab.com/libeigen/eigen eigen3
 WORKDIR /usr/src/eigen3/build/
 RUN cmake ..
-RUN make install -j4
+RUN make install -j2
 
 # Install TBB
 RUN apt-get -y install libtbb-dev
@@ -50,7 +50,7 @@ RUN cmake \
     ..
 
 # Build gtsam
-RUN make -j4 install && make clean
+RUN make -j2 install && make clean
 
 # Needed to link with GTSAM
 RUN echo 'export LD_LIBRARY_PATH=/usr/local/lib:LD_LIBRARY_PATH' >> /root/.bashrc
@@ -58,26 +58,26 @@ RUN echo 'export LD_LIBRARY_PATH=/usr/local/lib:LD_LIBRARY_PATH' >> /root/.bashr
 # install hdf5
 RUN apt-get -y install libhdf5-serial-dev
 
-# install highfive libraries
+# install HighFive v2.3.1
 WORKDIR /usr/src/
-RUN git clone --depth 1 --branch master https://github.com/BlueBrain/HighFive
+RUN git clone --depth 1 --branch v2.3.1 https://github.com/BlueBrain/HighFive
 WORKDIR /usr/src/HighFive/build/
 RUN cmake ..
-RUN make install -j4
+RUN make install -j2
 
 # install imuDataUtils
 WORKDIR /usr/src/
 RUN git clone --depth 1 --branch master https://github.com/tmcg0/imuDataUtils
 WORKDIR /usr/src/imuDataUtils/build/
 RUN cmake ..
-RUN make install -j4
+RUN make install -j2
 
 # install bioslam
 WORKDIR /usr/src/
 RUN git clone --depth 1 --branch dev https://github.com/tmcg0/bioslam
 WORKDIR /usr/src/bioslam/build/
 RUN cmake -DBIOSLAM_BUILD_MATLAB_WRAPPER=OFF ..
-RUN make install -j4
+RUN make install -j2
 
-# Run bash
-CMD ["bash"]
+# keep container running indefinitely
+ENTRYPOINT ["tail", "-f", "/dev/null"]

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -79,5 +79,4 @@ WORKDIR /usr/src/bioslam/build/
 RUN cmake -DBIOSLAM_BUILD_MATLAB_WRAPPER=OFF ..
 RUN make install -j2
 
-# keep container running indefinitely
-ENTRYPOINT ["tail", "-f", "/dev/null"]
+ENTRYPOINT ["/bin/bash"]

